tests/integration/test_argsort.py

```diff
@@ -0,0 +1,230 @@
+
+# ndarray.argsort(axis=-1, kind=None, order=None)
+
+DIM = 5
+SIZES = [
+    (0,),
+    (1),
+    (DIM),
+    (0, 1),
+    (1, 0),
+    (1, 1),
+    (1, DIM),
+    (DIM, 1),
+    (DIM, DIM),
+    (1, 0, 0),
+    (1, 1, 0),
+    (1, 0, 1),
+    (1, 1, 1),
+    (DIM, 1, 1),
+    (1, DIM, 1),
+    (1, 1, DIM),
+    (DIM, DIM, DIM),
+]
+
+NO_EMPTY_SIZES = [
+    (1),
+    (DIM),
+    (1, 1),
+    (1, DIM),
+    (DIM, 1),
+    (DIM, DIM),
+    (1, 1, 1),
+    (DIM, 1, 1),
+    (1, DIM, 1),
+    (1, 1, DIM),
+    (DIM, DIM, DIM),
+]
+
+SORT_TYPES = ["quicksort", "mergesort", "stable"]
+UNSTABLE_SORT_TYPE = ["heapsort"]
+
+
+class TestArgSort(object):
+    @pytest.mark.xfail
+    def test_arr_none(self):
+        res_np = np.argsort(
+            None
+        )  # numpy.AxisError: axis -1 is out of bounds for array of dimension 0
+        res_num = num.argsort(
+            None
+        )  # AttributeError: 'NoneType' object has no attribute 'shape'
+        assert np.equal(res_np, res_num)
+
+    @pytest.mark.parametrize("arr", ([], [[]], [[], []]))
+    def test_arr_empty(self, arr):
+        res_np = np.argsort(arr)
+        res_num = num.argsort(arr)
+        assert allclose(res_num, res_np)
+
+    @pytest.mark.xfail
+    def test_structured_array_order(self):
+        dtype = [("name", "S10"), ("height", float), ("age", int)]
+        values = [
+            ("Arthur", 1.8, 41),
+            ("Lancelot", 1.9, 38),
+            ("Galahad", 1.7, 38),
+        ]
+        a_np = np.array(values, dtype=dtype)
+        a_num = num.array(values, dtype=dtype)
+
+        res_np = np.argsort(a_np, order="height")
+        res_num = num.argsort(a_num, order="height")
+        # cuNumeric raises AssertionError in
+        # function cunumeric/cunumeric/eager.py:to_deferred_array
+        #     if self.deferred is None:
+        #         if self.parent is None:
+        #
+        # > assert self.runtime.is_supported_type(self.array.dtype)
+        # E
+        # AssertionError
+        #
+        # Passed on Numpy.
+        assert np.array_equal(res_np, res_num)
+
+        res_np = np.argsort(a_np, order=["age", "height"])
+        res_num = num.argsort(a_num, order=["age", "height"])
+        # same as above.
+        assert np.array_equal(res_np, res_num)
+
+    def test_axis_out_bound(self):
+        arr = [-1, 0, 1, 2, 10]
+        with pytest.raises(ValueError):
+            num.argsort(arr, axis=2)
+
+    @pytest.mark.xfail
+    def test_sort_type_invalid(self):
+        size = (3, 3, 2)
+        arr_np = np.random.randint(-3, 3, size)
+        arr_num = num.array(arr_np)
+        res_np = np.argsort(arr_np, kind="negative")
+        res_num = num.argsort(arr_num, kind="negative")
+        # Numpy raises "ValueError: sort kind must be one of 'quick',
+        # 'heap', or 'stable' (got 'negative')"
+        # cuNumeric passed. The code basically supports ‘stable’
+        # or not ‘stable’.
+        assert allclose(res_num, res_np)
```

Comment:
Why `allclose`? AFAIK `argsort` will return an array of integers, correct? (Here and below)

Reply:
change it to `assert np.array_equal`